### DISCUSSION TITLE
H1 - How to change the Battery Setting in HA

### DISCUSSION THREAD

**Discussion by gjhill6100** [nathanmarlor/foxess_modbus#263] [Q&A] [answered]
Sorry for the questions, but how do you set the following in HA with this Integration?

Max SOC
Period 1 : Start & End Times, Enable Charge from Grid and Enable Force Charge
Period 2 : Start & End Times, Enable Charge from Grid and Enable Force Charge

Looking for an easier way then having to use the Fox Cloud App for the times.
I current do not have the ability to set the Max SOC at the moment as Fox Cloud App or my User access does not allow me to do it.

I was hoping with this new Modbus integration would help me solve a lot of my issues.

The integration was easy to install one I got the WaveShare RS485 to ETH (B) working correctly. Well done for all of the hard work so far.

**Comment by canton7** [comment]
> Max SOC

Just edit the control: 

![image](https://github.com/nathanmarlor/foxess_modbus/assets/568104/8c81000e-218f-4290-83f1-bae242b949eb)

Or you can set it from an automation:

https://github.com/nathanmarlor/foxess_modbus/wiki/Setting-Inverter-modes-using-Automations

> Period 1 : Start & End Times, Enable Charge from Grid and Enable Force Charge

Developer tools -> Services -> "FoxESS - Modbus: Update Charge Period"

**Reply by gjhill6100** [reply]
I don’t have any options in the “Developer tools -> Services ->”

**Reply by canton7** [reply]
Screenshot?

**Reply by gjhill6100** [reply]
Found it....!

Didn't know the services were selected by a drop down menu.

Still reasonably new to Home Assistant and every days a school day :)

Many thanks for the responses.

**Reply by canton7** [reply, marked as answer]
I'm slowing writing a proper card you can put on a dashboard, which should make life a bit easier. It'll be done one day I'm sure!

**Reply by gjhill6100** [reply]
I will look forward to that :)

**Reply by smoketabs** [reply]
> I'm slowing writing a proper card you can put on a dashboard, which should make life a bit easier. It'll be done one day I'm sure!

I'll look forward to that, too :)

**Reply by azekeil** [reply]
I've created a PR to add these lifesaving instructions to the main README! #356

**Comment by calum-mcfarlane** [comment]
> I'm slowing writing a proper card you can put on a dashboard, which should make life a bit easier. It'll be done one day I'm sure!

You can (sort of) work around this by using the web page card to embed the URL for the HA Services page in a Lovelace dashboard. Obviously it will offer you all the other services defined in your HA install, but if you only ever manually activate the "FoxESS Modbus Update Charge Periods" service, you'll more-or-less have a card dedicated to that:
(Edit: Doesn't work on the app, but fine in a mobile browser)
![image](https://github.com/nathanmarlor/foxess_modbus/assets/115465775/b8e713c3-2198-4956-b7fa-ee3b67f191ea)

**Reply by canton7** [reply]
We're getting there... https://github.com/nathanmarlor/foxess_modbus_charge_period_card

You'll need foxess_modbus v1.7.0b2 or later. Add to HACS as Frontend -> Menu -> Custom repositories -> Enter the URL above, and choose "Lovelace".

![image](https://github.com/nathanmarlor/foxess_modbus/assets/568104/10a27e4b-5151-48bc-84c0-95e6a431bc59)

**Reply by FozzieUK** [reply]
very nice 👍

There's something a little bit out with the resizing, fine if i'm at full width but anything less the outer box resizes, and the inner is fixed

![image](https://github.com/nathanmarlor/foxess_modbus/assets/113460294/2c962c05-7e76-402b-bb76-350bdfc58826)

**Reply by canton7** [reply]
Fun, looks like it can't compress the times row, and, yeah, fails. I'll take a look at that next.

I'm no UI designer (if anything I'm the exact opposite), so if anyone has any UI suggestions please speak out!

**Reply by canton7** [reply]
Right, that's a slight improvement. I'll line it up a bit further for narrow layouts when I next get some time.

**Reply by FozzieUK** [reply]
That works perfectly for me, thanks 👍

**Comment by gjhill6100** [comment]
Hi All,

Is it the situation now that the "Enable charge period" option does not do
anything in the inverter?

If there is a time frame selected in either the Charge Period 1 or Charge
Period 2, then the Enable Charge is selected in the inverter.
If the "Enable charge from grid" is not selected, then the battery will not
be discharged during the selected period(s).
If the "Enable charge from grid" is selected, then the battery will charge
from the Grid during the selected period(s).

This corresponds to Fox changing their App with the "Enable charge period"
now disappearing from their App.

The time period needs to be able to be set with a start and end time as the
same, as this then disables the charge period within the Inverter.

Hopefully I have got the above correct as.

This is on the basis that I have been selecting the time frame and Force
from Grid on the Fox Ess App and then monitoring what has been selected /
set in the Modbus Interface and also what has been set within the Inverter
itself.

Kind Regards,

Graham Hill
There

**Reply by canton7** [reply]
The situation has *always* been that "Enable Charge period" is an application-level concept (the most recent mobile app scrapped this, but I'm keeping the concept in the HA integration as 1) it'd be a breaking, painful change to remove it, and 2) I think it's a useful concept).

You can enable/disable the "Enable Charge Period" toggle, and the correct things will be sent to the inverter.

**Comment by gjhill6100** [comment]
Hi,

No problems with 1) as doesn't do any halm (and less painful work for you)
:)

Does 2) still need the ability to be able to set the same start and end
times in the Card ?

I have tried various things and not been able to get the Service to
"disable" the charge window (i.e. get the start and end times the same in
the inverter to thus disable the feature).

Kind Regards

Graham

**Reply by canton7** [reply]
You disable the charge period with the service by unchecking the "Enable force charge" toggle, or passing `enable_force_charge: false`
